flagsmith/models.py

```diff
@@ -30,14 +28,18 @@ class Flag(BaseFlag):
     @classmethod
     def from_evaluation_result(
         cls,
-        flag: FlagResult,
-    ) -> Flag:
-        return Flag(
-            enabled=flag["enabled"],
-            value=flag["value"],
-            feature_name=flag["name"],
-            feature_id=int(flag["feature_key"]),
-        )
+        flag_result: SDKFlagResult,
+    ) -> typing.Optional[Flag]:
+        if (
+            flagsmith_id := (flag_result.get("metadata") or {}).get("flagsmith_id")
+        ) is not None:
+            return Flag(
+                enabled=flag_result["enabled"],
+                value=flag_result["value"],
+                feature_name=flag_result["name"],
+                feature_id=flagsmith_id,
+            )
+        return None
```

Comment:
```suggestion
    ) -> Flag:
        return Flag(
            enabled=flag_result["enabled"],
            value=flag_result["value"],
            feature_name=flag_result["name"],
            feature_id=flag_result["metadata"]["flagsmith_id"],
        )
```

If I understand correctly: this code path is only possible when evaluating a context mapped by this very client, meaning **there will be** metadata containing a Flagsmith ID. Also not sure what `return None` would mean here... It should never happen, right?

Reply:
Attempted to improve in d038a7383970f12c44b23bd3c11c1606cb1b6eec.